go/ephemeral/team_ek_box_storage.go

```diff
@@ -0,0 +1,374 @@
+	msg, _, err := userKeypair.DecryptFromString(teamEKBoxed.Box)
+	if err != nil {
+		return teamEK, err
+	}
+
+	seed, err := newTeamEKSeedFromBytes(msg)
+	if err != nil {
+		return teamEK, err
+	}
+
+	return keybase1.TeamEk{
+		Seed:     keybase1.Bytes32(seed),
+		Metadata: teamEKBoxed.Metadata,
+	}, nil
+}
+
+func (s *TeamEKBoxStorage) Put(ctx context.Context, teamID keybase1.TeamID, generation keybase1.EkGeneration, teamEKBoxed keybase1.TeamEkBoxed) (err error) {
+	defer s.G().CTrace(ctx, "TeamEKBoxStorage#Put", func() error { return err })()
+
+	s.Lock()
+	defer s.Unlock()
+
+	key, err := s.dbKey(ctx, teamID)
+	if err != nil {
+		return err
+	}
+	teamEKBoxes, _, err := s.getMap(ctx, teamID)
+	if err != nil {
+		return err
+	}
+	teamEKBoxes[generation] = teamEKBoxed
+	err = s.G().GetKVStore().PutObj(key, nil, teamEKBoxes)
+	if err != nil {
+		return err
+	}
+	s.cache.PutMap(teamID, teamEKBoxes)
+	return nil
+}
+
+func (s *TeamEKBoxStorage) Delete(ctx context.Context, teamID keybase1.TeamID, generation keybase1.EkGeneration) (err error) {
+	s.Lock()
+	defer s.Unlock()
+	return s.deleteMany(ctx, teamID, []keybase1.EkGeneration{generation})
+}
+
+func (s *TeamEKBoxStorage) deleteMany(ctx context.Context, teamID keybase1.TeamID, generations []keybase1.EkGeneration) (err error) {
+	defer s.G().CTrace(ctx, "TeamEKBoxStorage#delete", func() error { return err })()
+
+	teamEKBoxes, found, err := s.getMap(ctx, teamID)
+	if err != nil {
+		return err
+	} else if !found {
+		return nil
+	}
+
+	for _, generation := range generations {
+		delete(teamEKBoxes, generation)
+	}
+
+	key, err := s.dbKey(ctx, teamID)
+	if err != nil {
+		return err
+	}
+	err = s.G().GetKVStore().PutObj(key, nil, teamEKBoxes)
+	if err != nil {
+		return err
+	}
+	s.cache.PutMap(teamID, teamEKBoxes)
+	return nil
+}
+
+func (s *TeamEKBoxStorage) DeleteExpired(ctx context.Context, teamID keybase1.TeamID, merkleRoot libkb.MerkleRoot) (expired []keybase1.EkGeneration, err error) {
+	defer s.G().CTrace(ctx, "TeamEKBoxStorage#DeleteExpired", func() error { return err })()
+
+	s.Lock()
+	defer s.Unlock()
+
+	teamEKBoxes, found, err := s.getMap(ctx, teamID)
+	if err != nil {
+		return nil, err
+	} else if !found {
+		return nil, nil
+	}
+
+	for _, teamEKBox := range teamEKBoxes {
+		if ctimeIsStale(teamEKBox.Metadata.Ctime, merkleRoot) {
+			expired = append(expired, teamEKBox.Metadata.Generation)
+		}
+	}
+	return expired, s.deleteMany(ctx, teamID, expired)
+}
+
+func (s *TeamEKBoxStorage) GetAll(ctx context.Context, teamID keybase1.TeamID) (teamEKs TeamEKMap, err error) {
+	defer s.G().CTrace(ctx, "TeamEKBoxStorage#GetAll", func() error { return err })()
+
+	s.Lock()
+	defer s.Unlock()
+
+	teamEKs = make(TeamEKMap)
+	teamEKBoxes, found, err := s.getMap(ctx, teamID)
+	if err != nil {
+		return nil, err
+	} else if !found {
+		return teamEKs, nil
+	}
+
+	for generation, teamEKBoxed := range teamEKBoxes {
+		teamEK, err := s.unbox(ctx, teamEKBoxed)
+		if err != nil {
+			return teamEKs, err
```

Comment:
Slight preference for `return nil, err` in these branches.

Reply:
yes! forgot to switch those